src/JBMultiTerminal.sol

```diff
@@ -1587,25 +1648,23 @@ contract JBMultiTerminal is JBPermissioned, ERC2771Context, IJBMultiTerminal {
                         leftoverAmount -= amountFromFee;
                         returnedFees += feeAmount;
                     }
-                } else {
-                    // And here we overwrite with `feeAmountFrom` the `leftoverAmount`
-                    feeAmount = JBFees.feeAmountFrom({amount: leftoverAmount, feePercent: FEE});
 
+                    // Move the start index forward to the held fee after the current one.
+                    newStartIndex = startIndex + i + 1;
+                } else {
+                    // Get fee from `leftoverAmount`.
                     unchecked {
-                        _heldFeesOf[projectId][token].push(
-                            JBFee({
-                                amount: amountFromFee - leftoverAmount,
-                                beneficiary: heldFee.beneficiary,
-                                unlockTimestamp: heldFee.unlockTimestamp
-                            })
-                        );
-                        returnedFees += feeAmount;
+                        _heldFeesOf[projectId][token][startIndex + i].amount = heldFee.amount - leftoverAmount;
```

Comment:
this is the one line im not sure about. for some reason before it was `amountFromFee - leftoverAmount,` but i seems to me it should be `heldFee.amount - leftoverAmount`. need more eyes.

Reply:
Hmm yeah im also having a hard time reasoning about how it was before

Reply:
ok i got to the bottom of it by referencing v3. Neither were correct. @simplemachine92 do we have a test for this? if so it _must_ have been failing. 

Here's the case i just worked out in my notebook:

1. imagine a 25% fee.
2. a $100 payout would send $80 to the recipient and charge a $20 fee, since 20 is 25% of 80. the stored JBFee.amount is set to 100. paying back $80 should return the $20 fee.
3. lets say we add $75 back to balance. we need to get it so that adding $5 back to balance in a subsequent tx would clear up the fees cleanly.
4. during the first $75 addToBalance, `feeAmountIn(100)` is 20, so the amount paid out is 100-20 = 80. since 75 < 80, we go to the `else` block. `feeAmountFrom(75)` is $18.75. Meaning paying back $75 out of $80 should return $18.75 of fees out of $20. the right value to then decrement JBFee.amount is by $75 + $18.75 (the total paid back plus unlocked), which yields. $6.25.
5. during the next $5 addToBalance, `feeAmountIn(6.25)` is $1.25, so the paid amount is 6.25-1.25 = 5. since 5 == 5, we're done!

Reply:
this is where it apparently went off 
<img width="700" alt="image" src="https://github.com/user-attachments/assets/b90eeb45-83b2-40a5-a444-b27cae2348ef">
<img width="1386" alt="image" src="https://github.com/user-attachments/assets/5a0580f7-1adf-4906-b4b2-3cf4af5b7a5b">

jeez yall. @simplemachine92 how good to you feel about this test suite 😓